Cloud Infrastructure Security for AWS app (#3300)
SumoLogic/sumologic-documentation: docs/integrations/amazon-aws/cloud-infrastructure-security-for-aws.md

Key features of the app include:
* **Active threat monitoring**. See threats in APIs, resources, and storage.
* **Security compliance failure identification**. See areas in your environment that need to be addressed because they do not comply with required security standards.
* **Suspicious activity assessment**. See activity identified by anomaly detection across users, web interactions, networks, and Identity Access Management (IAM).
* **Risk overview**. See a summary of all resources that pose risks, and get an action plan for addressing the most important areas of concern.

Data presented in the app’s dashboards is normalized from log sources into AWS Elastic Common Schema (ECS) format, providing seamless data presentation of all your AWS data. 

Use Sumo Logic’s [monitoring](/docs/alerts/monitors/) to receive alerts from the app. To see monitors for the app, go to **Manage Data > Monitoring** and select the **Cloud Infrastructure Security** folder. 

## Log types

The Cloud Infrastructure Security for AWS app utilizes the following log types:

> * [Amazon Cloudtrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-getting-started.html) 

Reviewer: camelcase

```suggestion
* [Amazon CloudTrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-getting-started.html) 
```

| if(isEmpty(user.name), if(isEmpty(user_principal),"NA",user_principal), user.name) as user.name
// global filters
| where if ("*" = "*", true,user.name matches "*") AND if ("*" = "*", true, cloud.region matches "*") AND if ("*" = "*", true, cloud.account.id matches "*") AND if ("*" = "*", true, server.ip matches "*")
| where (event.action matches "ConsoleLogin" and error_message matches "Failed authentication")
// z-score calculation
| timeslice 3h
| count as eventCount by user.name, event.action, event_source, cloud.account.id, cloud.region, _timeslice
| sort + _timeslice
| rollingstd eventCount as eventCount_std by user.name, event.action, event_source, cloud.account.id, cloud.region
| smooth eventCount as eventCount_mean by user.name, event.action, event_source, cloud.account.id, cloud.region
| eventCount_std + 0.1 as eventCount_std
| (eventCount - eventCount_mean) / eventCount_std as zscore
| sort + _timeslice
| max(zscore) as max_zscore by user.name, event.action, event_source, cloud.account.id, cloud.region
| round(max_zscore, 2) as max_zscore
| where max_zscore > "1"
| sort - max_zscore
```

## Collecting logs for Cloud Infrastructure Security for AWS

> The app collects logs from different AWS sources to produce data in the dashboards. When you install the app, data will be collected from sources including:

Reviewer: ```suggestion
The app collects logs from different AWS sources to produce data in the dashboards. When you install the app, data will be collected from sources, including:
```

> * [Amazon Cloudtrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-getting-started.html) 

Reviewer: ```suggestion
* [Amazon CloudTrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-getting-started.html) 
```

   1. For CloudTrail configuration:
      * Under **5.1 CloudTrail service configuration**, in **CloudTrail Regions**, enter regions from which CloudTrail Data should be sent.
      * Under **5.2 CloudTrail Sumo log source configuration**, under **Create Sumo Logic S3 logs source for CloudTrail**, choose **Yes** to create the Sumo Logic S3 log source to collect CloudTrail logs, or choose **No** to skip creation.
      * **Path expression for logs**. The path expression must match the folder structure for CloudTrail logs (for example, `AWSLogs/*/CloudTrail/*`).
      * **Sumo Logic CloudTrail logs source category name**. Required when the flag is set to **No**. Provide the name of an existing Sumo Logic source category that's collecting CloudTrail logs. This is also used for Threat Intel for AWS app installation.
      * Under **5.3 CloudTrail S3 bucket configuration**, in **Create an S3 bucket for CloudTrail logs**, choose **Yes** to create an S3 bucket for CloudTrail logs.

>       * **Name of existing S3 bucket that contains the CloudTrail logs**. Provide the name of an existing S3 bucket that contains CloudTrail logs. Required when the flag is set to **No**.  The existing bucket must be in same AWS Region as the log-archiving account. 

Reviewer: ```suggestion
      * **Name of existing S3 bucket that contains the CloudTrail logs**. Provide the name of an existing S3 bucket that contains CloudTrail logs. Required when the flag is set to **No**. The existing bucket must be in same AWS Region as the log-archiving account. 
```

      * **Delivery bucket prefix**. Enter the log delivery S3 bucket prefix. <br/><img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-param-5.png')} alt="CloudTrail configuration" style={{border: '1px solid black'}} width="700"/> 
   1. For Security Hub configuration:
      * Under **6.1 Security Hub Service Configuration**, in **Security Hub Regions**, enter regions from which Security Hub Data should be sent.

>       * Undr **6.2 Security Hub Sumo Log Source configuration**, in **Create Sumo Logic HTTP logs source**, select **Yes** to create Sumo Logic HTTP log source to collect Security Hub logs, or select **No** to skip creation.

Reviewer: ```suggestion
      * Under **6.2 Security Hub Sumo Log Source configuration**, in **Create Sumo Logic HTTP logs source**, select **Yes** to create Sumo Logic HTTP log source to collect Security Hub logs, or select **No** to skip creation.
```

1. Click **Create Stack**. The stack is created, and the app is installed.
1. Select an option to start using the app. <br/><img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-finish-installation.png')} alt="App hub page" style={{border: '1px solid black'}} width="700"/>
   

## Cloud Infrastructure Security for AWS app dashboards​

{@import ../../reuse/filter-dashboards.md}

### Active threats dashboards

The Active Threats dashboards show data on threats that require attention. Review these dashboards to see threats identified in AWS APIs, resources, and storage.

#### Active Threats: AWS APIs

The **Active Threats: AWS APIs** dashboard shows threats identified from AWS APIs. It shows threats count and trend, and threats by resource, actor, events, and geo location. 

<img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-active-threats-aws-apis.png')} alt="Active Threats: AWS APIs dashboard" width="600"/>

#### Active Threats: AWS Resources

> The **Active Threats: AWS Resources** dashboard shows threats identified in AWS resources such as EC2 and IAMUser. It shows findings by resource, trend, resource type, category, and country. Like the [Risk Overview](#resource-risk-dashboard) dashboard, this dashboard has an **Action Plan** panel so you can access suggested resources that need attention.

Reviewer: ```suggestion
The **Active Threats: AWS Resources** dashboard shows threats identified in AWS resources such as EC2 and IAMUser. It shows findings by resource, trend, resource type, category, and country. This dashboard has an **Action Plan** panel so you can access suggested resources that need attention. 
```

<img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-active-threats-aws-resources.png')} alt="Active Threats: AWS APIs dashboard" width="600"/>

#### Active Threats: AWS Storage

The **Active Threats: AWS Storage** dashboard provides threat counts related to AWS S3 buckets. It shows threats count and trend, and threats by resource, actor, and geo location.

<img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-active-threats-aws-storage.png')} alt="Active Threats: AWS Storage dashboard" width="600"/>

### Risk Overview dashboard

> The **Risk Overview** dashboard provides a summary of all resources that pose risks in a single dashboard that rolls up the findings from other dashboards. It also shows AWS API events by time, and has an **Action Plan** panel so you can access resources that need attention.

Reviewer: ```suggestion
The **Risk Overview** dashboard provides a summary of all resources that pose risks in a single dashboard that rolls up the findings from other dashboards. It also shows AWS API events by time. Like the [Active Threats: AWS Resources](#active-threats-aws-resources) dashboard, this dashboard has an **Action Plan** panel so you can access resources that need attention.
```

Author: @kimsauce, I won't make this change, because it's not needed.

---
id: cloud-infrastructure-security-for-aws
title: Cloud Infrastructure Security for AWS
description: The Cloud Infrastructure Security for AWS app provides visibility into your AWS environment to give you insights into active threats, security control failures, and suspicious activity. 
---

import useBaseUrl from '@docusaurus/useBaseUrl';

> <img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-logo.png')} alt="Cloud Infrastructure Security for AWS logo" width="100"/>

Reviewer: looks a little big

```suggestion
<img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-logo.png')} alt="Cloud Infrastructure Security for AWS logo" width="70"/>
```

Data presented in the app’s dashboards is normalized from log sources into AWS Elastic Common Schema (ECS) format, providing seamless data presentation of all your AWS data. 

Use Sumo Logic’s [monitoring](/docs/alerts/monitors/) to receive alerts from the app. To see monitors for the app, go to **Manage Data > Monitoring** and select the **Cloud Infrastructure Security** folder. 

## Log types

The Cloud Infrastructure Security for AWS app utilizes the following log types:
* [Amazon CloudTrail](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-getting-started.html) 
* [Amazon GuardDuty findings](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_findings.html)
* [AWS Network Firewall](https://docs.aws.amazon.com/network-firewall/latest/developerguide/what-is-aws-network-firewall.html)
* [AWS Security Hub](https://docs.aws.amazon.com/securityhub/latest/userguide/what-is-securityhub.html)
* [AWS Web Application Firewall (WAF)](https://docs.aws.amazon.com/waf/latest/developerguide/waf-chapter.html) 

## Sample log message

> ```

Reviewer: ```suggestion
```json
```

{ 
  "eventVersion":"1.01",
  "userIdentity":{ 
     "type":"IAMUser",
     "principalId":"AIDA4XQZKIVURYEOA",
     "arn":"arn:aws:iam::95619384238:user/Olaf",
     "accountId":"95238468",
     "userName":"system"
  },
  "eventTime":"2017-09-27T20:00:10Z",
  "eventSource":"signin.amazonaws.com",
  "eventName":"ConsoleLogin",
  "awsRegion":"us-east-1",
  "sourceIPAddress":"192.0.2.0",
  "userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/37.0.2062.124 Safari/537.36",
  "requestParameters":null,
  "responseElements":{ 
     "ConsoleLogin":"Failure"
  },
  "additionalEventData":{ 
     "MobileVersion":"No",
     "LoginTo":"https://console.aws.amazon.com/console/home?state\u003dhashArgs%23\u0026isauthcode\u003dtrue",
     "MFAUsed":"No"
  },
  "eventID":"f36c1d07-73cf-4ab8-84b1-04c93ad3aaeb"
}
```

## Sample query

##### Failed Console Logins

> ```

Reviewer: ```suggestion
```sql
```

      * **Sumo Logic Amazon S3 Logs Source Category Name for Network Firewall**. Enter the name of a source category from Sumo Logic if it already exists. To create a new source category, use the default name provided.
      * Under **7.4 Firewall Manager - S3 Bucket Configuration**, in **Create AWS S3 Bucket**, select **Yes** to create a new S3 bucket in AWS S3. Select **No** to use an existing S3 bucket from AWS S3 which has Network Firewall Logs.
      * **Network Firewall Delivery Bucket Prefix**. Enter the Network Firewall Log Delivery S3 bucket prefix.
      * **Name of existing S3 Bucket which contains the Network Firewall Logs**. Provide an existing S3 Bucket name which contains Network Firewall Logs. Required when flag is set to **No**. <br/><img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-param-7.png')} alt="Firewall configuration" style={{border: '1px solid black'}} width="700"/> 
   1. For Quick Start configuration:

>       * Under **8. AWS Quick Start configuration**, in **Quick Start S3 bucket name**, enter the name of the S3 bucket for your copy of the Quick Start assets. Keep the default name unless you are customizing the template. Changing the name updates code references to point to a new Quick Start location. This name can include numbers, lowercase letters, uppercase letters, and hyphens, but do not start or end with a hyphen (-). See https://aws-quickstart.github.io/option1.html.
>       * **Quick Start S3 key prefix**. Enter the S3 key prefix that is used to simulate a directory for your copy of the Quick Start assets. Keep the default prefix unless you are customizing the template. Changing this prefix updates code references to point to a new Quick Start location. This prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), and forward slashes (/). End with a forward slash. See https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html and https://aws-quickstart.github.io/option1.html.

Reviewer: ```suggestion
      * Under **8. AWS Quick Start configuration**, in **Quick Start S3 bucket name**, enter the name of the S3 bucket for your copy of the Quick Start assets. Keep the default name unless you are customizing the template. Changing the name updates code references to point to a new Quick Start location. This name can include numbers, lowercase letters, uppercase letters, and hyphens, but do not start or end with a hyphen (-). See [Adapt a CloudFormation-based Partner Solution](https://aws-quickstart.github.io/option1.html).
      * **Quick Start S3 key prefix**. Enter the S3 key prefix that is used to simulate a directory for your copy of the Quick Start assets. Keep the default prefix unless you are customizing the template. Changing this prefix updates code references to point to a new Quick Start location. This prefix can include numbers, lowercase letters, uppercase letters, hyphens (-), and forward slashes (/). End with a forward slash. See [Working with object metadata](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html) and [Adapt a CloudFormation-based Partner Solution](https://aws-quickstart.github.io/option1.html).
```

:::info
**Preview**<br/>The Cloud Security Infrastructure for AWS solution is in a preview release, and has the following limitations:
* The CloudFormation template only supports one AWS region (with multiple AWS accounts) during preview. You are encouraged to get started with one AWS region. 
* Pre-built monitors are not installed.
:::

Cloud Infrastructure Security for AWS provides a unified view of threats, misconfigurations, and threats in your AWS infrastructure spanning multiple AWS accounts and regions. The solution leverages native AWS tools and telemetry to accelerate cloud security outcomes.

Key features of the solution include:
* **Risk overview**. See a summary of all resources that pose risks, and get an action plan for addressing the most important areas of concern.
* **Active threats**. See threats in resources and AWS API calls.
* **Security control failures**. See misconfigurations  in your environment that may leave you vulnerable to attackers.
* **Suspicious activity**. See activity identified by anomaly detection across users, web interactions, networks, and Identity Access Management (IAM).

> Entities  presented in the dashboards are normalized from log sources into [AWS Elastic Common Schema (ECS)](https://www.elastic.co/guide/en/ecs/master/ecs-reference.html), to provide seamless pivots between dashboards during threat investigations. 

Reviewer: ```suggestion
Entities presented in the dashboards are normalized from log sources into [AWS Elastic Common Schema (ECS)](https://www.elastic.co/guide/en/ecs/master/ecs-reference.html), to provide seamless pivots between dashboards during threat investigations. 
```

   * [AWS Web Application Firewall (WAF)](https://docs.aws.amazon.com/waf/latest/developerguide/waf-chapter.html) 
* **Sumo Logic console**. Make sure you have access to the [Sumo Logic console](/docs/get-started/sumo-logic-ui/) and as a user that is associated with Sumo Logic role and required role capabilities.
* **Role capabilities**. Make sure you have a Sumo Logic role that have the following [role capabilities](/docs/manage/users-roles/roles/role-capabilities/):
  * Manage field extraction rules
  * Manage connections
  * View Account Overview
  * View Fields
  * View field extraction rules
  * Manage Content
  * Manage Collectors
  * View Collectors
  * Manage Fields
  * Manage Monitors
  * Manage Metrics Rules
  * View Monitors
  * Manage Entity Type Configs
  * Create access keys
* **Sumo Logic Access ID and Key**. When you deploy the solution, you’ll need to supply a Sumo Logic [Access ID and Access Key](/docs/manage/security/access-keys/), which enable you to use Sumo Logic APIs. Make sure you have the role capabilities listed above before generating the Access ID and Key.

> * **AWS credentials**. To deploy the solution, you will need to log onto the [AWS Console]([AWS Console](https://console.aws.amazon.com/console/)). As necessary, you may add JSON text to an existing or a new policy associated with an AWS IAM role as described in the [AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-and-attach-iam-policy.html). 

Reviewer: ```suggestion
* **AWS credentials**. To deploy the solution, you will need to log onto the [AWS Console](https://console.aws.amazon.com/console). As necessary, you may add JSON text to an existing or a new policy associated with an AWS IAM role as described in the [AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-and-attach-iam-policy.html). 
```

> * **Monitors**. The Cloud Infrastructure Security for AWS solution comes with pre-packaged alerts in the form of Sumo Logic Monitors. To understand more about their capabilities please visit the [Monitors](/docs/alerts/monitors/) page.

Reviewer: ```suggestion
* **Monitors**. The Cloud Infrastructure Security for AWS solution comes with pre-packaged alerts in the form of Sumo Logic Monitors. To learn more about their capabilities, visit the [Monitors](/docs/alerts/monitors/) page.
```

   1. Click **Create Stack**. The stack is created, and the solution is installed.
1. Click **Start Using Sumo**. <br/><img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-start-using-sumo.png')} alt="Start Using Sumo button" style={{border: '1px solid black'}} width="400"/>
1. Select an option to start using the solution. <br/><img src={useBaseUrl('img/integrations/amazon-aws/cis-for-aws-finish-installation.png')} alt="App hub page" style={{border: '1px solid black'}} width="800"/>
   
### Troubleshoot installation

Installation of Cloud Infrastructure Security for AWS uses an AWS CloudFormation template. While deploying using the template, you may receive error messages such as `CREATE_FAILED` status or `ROLLBACK_COMPLETE` status for various reasons. This section provides information on how to troubleshoot such AWS CloudFormation installation failures.

#### Determine the cause of a CloudFormation installation failure

This section walks you through the process of troubleshooting an AWS CloudFormation installation failure.

To debug an AWS CloudFormation installation failure, do the following:

> 1. After the stack rollback is complete and the status is ROLLBACK_COMPLETE, go to the parent stack. In the parent stack, look for the first failure as shown in the following example. The failure can be a direct reason or can point to a nested stack. <br/><img src={useBaseUrl('img/observability/Troubleshooting_1.png')} alt="Troubleshooting 1" style={{border: '1px solid black'}} width="800"/>

Reviewer: ```suggestion
1. After the stack rollback is complete and the status is `ROLLBACK_COMPLETE`, go to the parent stack. In the parent stack, look for the first failure as shown in the following example. The failure can be a direct reason or can point to a nested stack. <br/><img src={useBaseUrl('img/observability/Troubleshooting_1.png')} alt="Troubleshooting 1" style={{border: '1px solid black'}} width="800"/>
```

1. Look for direct reasons for the failure that is available in the parent stack, as shown in the following example. <br/><img src={useBaseUrl('img/observability/Troubleshooting_2.png')} alt="Troubleshooting 2" style={{border: '1px solid black'}} width="800"/>
1. To find indirect reasons for the failure, go to the nested stack mentioned in the status reason, as shown in the following example. Take a note of the resources mentioned in the reason. <br/><img src={useBaseUrl('img/observability/Troubleshooting_3.png')} alt="Troubleshooting 3" style={{border: '1px solid black'}} width="800"/>

> 1. Select the deleted option to find the nested stacks, as shown in the following example.<br/><img src={useBaseUrl('img/observability/Troubleshooting_4.png')} alt="Troubleshooting 4" style={{border: '1px solid black'}} width="500"/>

Reviewer: I'd shrink this a bit because the image is a little pixelated

```suggestion
1. Select the deleted option to find the nested stacks, as shown in the following example.<br/><img src={useBaseUrl('img/observability/Troubleshooting_4.png')} alt="Troubleshooting 4" style={{border: '1px solid black'}} width="400"/>
```
